cmd/tcpgoon.go

```diff
@@ -53,26 +57,65 @@ func init() {
 	runCmd.Flags().BoolVarP(&params.debug, "debug", "d", false, "Print debugging information to the standard error")
 	runCmd.Flags().IntVarP(&params.reportingInterval, "interval", "i", 1, "Interval, in seconds, between stats updates")
 	runCmd.Flags().BoolVarP(&params.assumeyes, "assume-yes", "y", false, "Force execution without asking for confirmation")
+	runCmd.Flags().BoolVarP(&params.docker, "docker", "", false, "Running against a docker image. " +
+	"Use the image reference instead of the <host>, and a custom mapped port if the first exposed one is not ok (optional")
 }
 
 func validateRequiredArgs(params *tcpgoonParams, args []string) error {
```

Comment:
we must redo how we are validating parms. Relying on a max/min num of required params looks ugly: we should just check which parm is required and which are not in each execution mode. If execution modes parameters differ a lot in the end, maybe we should recover your idea of using different commands

Reply:
yes, the execution mode is the first question to answer